scripts/gen_checkerboard.py

```diff
@@ -0,0 +1,194 @@
+logging.getLogger().setLevel(logging.DEBUG)
+# logging.getLogger().setLevel(logging.WARNING)
+import argparse
+
+parser = argparse.ArgumentParser()
+
+""" useful information when running from a GIT folder."""
+myname = os.path.realpath(__file__)
+mypath = os.path.dirname(myname)
+myname = os.path.basename(myname)
+
+
+def make_box(n_inlines, n_crosslines, n_depth, box_size):
+    """
+    Makes a 3D box in checkerboard pattern.
+
+    :param n_inlines: dim x
+    :param n_crosslines: dim y
+    :param n_depth: dim z
+    :return: numpy array
+    """
+    # inline by crossline by depth
+    zero_patch = np.ones((box_size, box_size)) * WHITE
+    one_patch = np.ones((box_size, box_size)) * BLACK
+
+    # multiplying by one flips the sign
+    stride = np.hstack((zero_patch, one_patch))
+    neg_stride = -1 * stride
+
+    # come up with a 2D inline image
+    nx, ny = stride.shape
+
+    step_col = int(np.ceil(n_crosslines / float(ny)))
+    step_row = int(np.ceil(n_inlines / float(nx) / 2))
+
+    # move in the crossline direction
+    crossline_band = np.hstack((stride,) * step_col)
+    neg_crossline_band = -1 * crossline_band
+
+    checker_stripe = np.vstack((crossline_band, neg_crossline_band))
+
+    # move down, alternative
+    checker_image = np.vstack((checker_stripe,) * step_row)
+
+    # trim excess
+    checker_image = checker_image[0:n_inlines, 0:n_crosslines]
+
+    # now make a box with alternating checkers
+    checker_box = np.ones((n_inlines, n_crosslines, box_size * 2))
+    checker_box[:, :, 0:box_size] = checker_image[:, :, np.newaxis]
+    # now invert the colors
+    checker_box[:, :, box_size:] = -1 * checker_image[:, :, np.newaxis]
+
+    # stack boxes depth wise
+    step_depth = int(np.ceil(n_depth / float(box_size) / 2))
+    final_box = np.concatenate((checker_box,) * step_depth, axis=2)
+
+    return final_box
+
+
+def mkdir(path):
+    """
+  Create a directory helper function
+  """
+    if not os.path.isdir(path):
+        os.mkdir(path)
+
+
+def main(args):
+    """
+
+    Generates checkerboard dataset based on Dutch F3 in Alaudah format.
+
+    TODO: please see the TODO statements in the code
+
+    """
+
+    logging.info("loading data")
+
+    train_seismic = np.load(os.path.join(args.dataroot, "train", "train_seismic.npy"))
+    train_labels = np.load(os.path.join(args.dataroot, "train", "train_labels.npy"))
+    test1_seismic = np.load(os.path.join(args.dataroot, "test_once", "test1_seismic.npy"))
+    test1_labels = np.load(os.path.join(args.dataroot, "test_once", "test1_labels.npy"))
+    test2_seismic = np.load(os.path.join(args.dataroot, "test_once", "test2_seismic.npy"))
+    test2_labels = np.load(os.path.join(args.dataroot, "test_once", "test2_labels.npy"))
+
+    assert train_seismic.shape == train_labels.shape
+    assert train_seismic.min() == WHITE
+    assert train_seismic.max() == BLACK
+    assert train_labels.min() == 0
+    assert train_labels.max() == 5
```

Comment:
Are 0 and 5 here related to the number of classes in the Dutchf3 datsset?

Reply:
Yes, added comment to address this